src/clr-angular/data/datagrid/render/cell-renderer.ts

```diff
@@ -4,15 +4,32 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 import { Directive, ElementRef, OnDestroy, Renderer2 } from '@angular/core';
-import { Subscription } from 'rxjs';
+import { BehaviorSubject, Subscription } from 'rxjs';
 
 import { DatagridRenderStep } from '../enums/render-step.enum';
 
 import { STRICT_WIDTH_CLASS } from './constants';
 import { DatagridRenderOrganizer } from './render-organizer';
+import { DatagridColumnState } from '../interfaces/column-state.interface';
+import { DatagridColumnChanges } from '../enums/column-changes.enum';
 
 @Directive({ selector: 'clr-dg-cell' })
 export class DatagridCellRenderer implements OnDestroy {
+  stateSubscription: Subscription;
+  _columnState: BehaviorSubject<DatagridColumnState>;
+
+  get columnState(): BehaviorSubject<DatagridColumnState> {
+    return this._columnState;
+  }
+
+  set columnState(columnState: BehaviorSubject<DatagridColumnState>) {
+    if (this.stateSubscription) {
+      this.stateSubscription.unsubscribe();
+    }
+    this._columnState = columnState;
+    this.stateSubscription = columnState.asObservable().subscribe(state => this.stateChanges(state));
```

Comment:
`asObservable()` doesn't do anything here. It's only useful for type safety if you want to expose a public `Observable` and keep a private `Subject`. But the code for `asObservable()` in RxJS is really just:
```typescript
asObservable() {
  return this as Observable<T>;
}
```